### PR TITLE
Add Cell check to RadarWidget

### LINKED ISSUE #9033: Playtest skirmish crash
playtest-20150808

my orca and an AI orca killed each other, then the AI initiated an airstrike. Game was unresponsive for a few seconds, then crashed back to desktop (win7)

replay file: http://www.pipelinestall.org/stuff/OpenRA-2015-08-11T094144Z.orarep

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25644206-playtest-skirmish-crash?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by LipkeGu**
Fixes #9033

### CODE REVIEW

**Comment by Mailaender** [comment]
:+1:

File: OpenRA.Mods.Common/Widgets/RadarWidget.cs [hunk 1]
```diff
@@ -85,7 +85,8 @@ public override void Initialize(WidgetArgs args)
 
 			// Set initial terrain data
 			foreach (var cell in world.Map.AllCells)
-				UpdateTerrainCell(cell);
+				if (world.Map.Contains(cell))
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/RadarWidget.cs]: Map.AllCells by definition only contains cells in the map.  This check is unnecessary and just shows down the game.

File: OpenRA.Mods.Common/Widgets/RadarWidget.cs [hunk 2]
```diff
@@ -278,7 +279,9 @@ public override void Draw()
 			if (renderShroud != null)
 			{
 				foreach (var cell in dirtyShroudCells)
-					UpdateShroudCell(cell);
+					if (world.Map.Contains(cell))
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/RadarWidget.cs]: Cells are (or should be) checked before they are added to dirtyShroudCells, so this is also unnecessary.

**Comment by LipkeGu** [comment]
updated

**Comment by RoosterDragon** [comment]
:+1:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/b2a3348d602248b46a28a3a8ba4f8af16737947f...3ee6ea5bfb348d368c9de31b9de4dfaca39839ee)

**Pull request merged**